### PR TITLE
Cursor should not change for other teams when hovering over doors and construction.

### PR DESCRIPTION
**Pull request by MEEPofFaith**
Before:

https://user-images.githubusercontent.com/54301439/109609374-1942c000-7ae0-11eb-83d4-1b25365abcd3.mp4

Mindustry_exPb9xVUee

After:

https://user-images.githubusercontent.com/54301439/109609388-1d6edd80-7ae0-11eb-89fa-57a7c100ce8a.mp4

java_lVBIYnJCzn

### CODE REVIEW

**Comment by MEEPofFaith** [comment]
(I need to fix the recording of my cursor...)

File: core/src/mindustry/world/blocks/defense/Door.java [hunk 1]
```diff
@@ -126,7 +126,7 @@ public void draw(){
 
         @Override
         public Cursor getCursor(){
-            return SystemCursor.hand;
+            return player.team() == team ? SystemCursor.hand : SystemCursor.arrow;
```
Comment (Quezler) [inline core/src/mindustry/world/blocks/defense/Door.java:129]: it should probably use one of the built-in interaction checks:

https://github.com/Anuken/Mindustry/search?q=canInteract&type=

```suggestion
            return interactable(player.team()) ? SystemCursor.hand : SystemCursor.arrow;
```

**Comment by MEEPofFaith** [comment]
Wait why does Door even override `getCursor` if it already is correct in `building`.
image

**Comment by MEEPofFaith** [comment]
Wait that's because doors aren't configurable, just clickable.

**Review by Quezler — APPROVED** [review]

**Pull request merged**